src/EditorFeatures/Core/Implementation/Diagnostics/AbstractDiagnosticsTaggerProvider.AggregatingTagger.cs

```diff
@@ -137,6 +138,15 @@ private void OnDiagnosticsUpdated(object sender, DiagnosticsUpdatedArgs e)
                 ValueTuple<TaggerProvider, IAccurateTagger<TTag>> providerAndTagger;
                 if (!_idToProviderAndTagger.TryGetValue(id, out providerAndTagger))
                 {
+                    // We didn't have an existing tagger for this diagnostic id.  If there are no actual 
+                    // diagnostics being reported, then don't bother actually doing anything.  This saves
+                    // us from creating a lot of objects, and subscribing to tons of events that we don't
+                    // actually need (since we don't even have any diagnostics to show!).
+                    if (e.Diagnostics.Length == 0)
```

Comment:
diagnostic that are reported and diagnostic each tagger cares are different. so, probably need to filter these to see whether there is any diagnostic actually  we care.